### PR TITLE
add indonesia holiday

### PR DESCRIPTION
**Pull request by ddouble-d**
Added regular holidays https://publicholidays.co.id/2024-dates

### CODE REVIEW

File: src/Countries/Indonesia.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+use Carbon\CarbonImmutable;
+
+class Indonesia extends Country
+{
+    public function countryCode(): string
+    {
+        return 'id';
+    }
+
+    /** @return array<string, CarbonImmutable> */
```
Comment (Nielsvanpach) [inline src/Countries/Indonesia.php:14]: You can remove this

**Review by Nielsvanpach — COMMENTED** [review]
Thanks! Aren't there any official Islamic public holidays in Indonesia?

**Comment by ddouble-d** [comment]
> Thanks! Could you sync your fork with this repo and rebase your branch on our main branch? That way I can run the tests here.

Sure


> Also, aren't there any official Islamic public holidays in Indonesia?

Actually there is, but need to add some method/trait that can be used also for other countries.
For example islamic calendar will be used also for malaysia / other islamic countries.

**Comment by Nielsvanpach** [comment]
Is the Chinese Calendar used in Indonesia for public holidays? Chinese New Year for example?
https://github.com/spatie/holidays/blob/main/src/Calendars/ChineseCalendar.php

**Comment by ddouble-d** [comment]
> Is the Chinese Calendar used in Indonesia for public holidays? Chinese New Year for example? https://github.com/spatie/holidays/blob/main/src/Calendars/ChineseCalendar.php

yep, updated the pr

File: tests/Countries/IndonesiaTest.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Spatie\Holidays\Tests\Countries;
+
+use Carbon\CarbonImmutable;
+use Spatie\Holidays\Holidays;
+
+it('can calculate Indonesia holidays', function () {
+    CarbonImmutable::setTestNowAndTimezone('2024-01-01');
```
Comment (Nielsvanpach) [inline tests/Countries/IndonesiaTest.php:9]:
```suggestion
    CarbonImmutable::setTestNow('2024-01-01');
```

**Review by Nielsvanpach — APPROVED** [review]
👍

File: tests/Countries/IndonesiaTest.php (hunk 2 above)
Comment (ddouble-d) [reply tests/Countries/IndonesiaTest.php]: done 👍

**Pull request merged**